### ISSUE TITLE
IPv6 service throw ServiceIPAddressCannotBeChangedOnUpdate error when ensure load balancer.

### ISSUE THREAD

**Issue by TonyDou** [Azure/aks-engine#4121]
**Describe the bug**
If the static IP for the loadbalancer service has the property servicePublicIPAddress, it will fail to ensure load balancer when creating the service with an existing static IP.
```
  Normal   EnsuringLoadBalancer           4m54s (x80 over 6h15m)  service-controller    Ensuring load balancer
  Warning  CreateOrUpdatePublicIPAddress  4m53s (x80 over 6h15m)  azure-cloud-provider  Retriable: false, RetryAfter: 0s, HTTPStatusCode: 400, RawError: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 400, RawError: {
  "error": {
    "code": "ServiceIPAddressCannotBeChangedOnUpdate",
    "message": "Service IP Address property is set at creation time and cannot be changed.",
    "details": []
  }
}
```

**Steps To Reproduce**
1. Deploy cluster with dual stack enabled.
```json
{
    "apiVersion": "vlabs",
    "location": "southcentralus",
    "properties": {
        "featureFlags": {
            "enableIPv6DualStack": true
        },
        "orchestratorProfile": {
            "orchestratorType": "Kubernetes",
            "orchestratorRelease": "1.20",
            "kubernetesConfig": {
                "networkPlugin": "kubenet",
                "kubeProxyMode":"ipvs",
                "useManagedIdentity": true,
                "cloudProviderDisableOutboundSNAT": true,
                "apiServerConfig": {
                    "--feature-gates": "IPv6DualStack=true"
                },
                "kubeletConfig": {
                    "--feature-gates": "IPv6DualStack=true"
                },
                "controllerManagerConfig": {
                    "--feature-gates": "IPv6DualStack=true"
                }
            }
        },
        "masterProfile": {
            "count": 1,
            "dnsPrefix": "***",
            "vmSize": "Standard_D2_v3"
        },
        "agentPoolProfiles": [
            {
                "name": "linuxpool1",
                "count": 2,
                "vmSize": "Standard_D2_v3",
                "availabilityProfile": "VirtualMachineScaleSets"
            }
        ],
        "linuxProfile": {
            "adminUsername": "***",
            "ssh": {
                "publicKeys": [
                    {
                        "keyData": "****"
                    }
                ]
            }
        }
    }
}
```
2. Create an IPv6 address with servicePublicIPAddress property. To create IPv6 address with this property, you also have to create an IPv4 address.
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {},
    "resources": [
        {
            "name": "test-ipv4",
            "type": "Microsoft.Network/publicIPAddresses",
            "apiVersion": "2020-05-01",
            "location": "[resourceGroup().location]",
            "sku": {
                "name": "Standard"
            },
            "properties": {
                "publicIPAllocationMethod": "Static",
                "publicIPAddressVersion": "IPv4",
                "IpTags": ["***IpTags****" ]
            }
        },
        {
            "name": "test-ipv6",
            "type": "Microsoft.Network/publicIPAddresses",
            "apiVersion": "2020-05-01",
            "location": "[resourceGroup().location]",
            "sku": {
                "name": "Standard"
            },
            "properties": {
                "publicIPAllocationMethod": "Static",
                "publicIPAddressVersion": "IPv6",
                "servicePublicIPAddress": {
                    "id": "[resourceid('Microsoft.Network/publicIPAddresses/', 'test-ipv4')]"
                }
            },
            "dependsOn": [
                "[resourceid('Microsoft.Network/publicIPAddresses/', 'test-ipv4')]"
            ]
        }
    ],
    "outputs": {
        "IPs": {
            "type": "array",
            "value": [
                "[reference(resourceid('Microsoft.Network/publicIPAddresses/', 'test-ipv4')).ipAddress]",
                "[reference(resourceid('Microsoft.Network/publicIPAddresses/', 'test-ipv6')).ipAddress]"
            ]
        }
    }
}
```
3. Create an IPv6 loadbalancer service to reference the IPv6 address created in step 2.
```yaml
apiVersion: v1
kind: Service
metadata:
  name: nginx-se-ipv6
spec:
  ipFamilies:
    - IPv6
  externalTrafficPolicy: Local
  selector:
    app: nginx
  ports:
    - protocol: TCP
      port: 80
      targetPort: 80
  type: LoadBalancer
  loadBalancerIP: <***the-ipv6-address-created-in-step-2***>
```
It will pending on ensuring load balancer, and never succeed. When describe the loadbalancer service, it shows following error message. From the error message, looks like it's trying to update the IPv6 address without the servicePublicIPAddress property while ensuring load balancer.
```
  "error": {
    "code": "ServiceIPAddressCannotBeChangedOnUpdate",
    "message": "Service IP Address property is set at creation time and cannot be changed.",
    "details": []
  }
```
**Expected behavior**
It should successfully ensure the load balancer with a specified static IPv6 address with property servicePublicIPAddress.

**AKS Engine version**
v0.58.0

**Kubernetes version**
1.20

**Additional context**

**Comment by TonyDou** [comment]
The issue is actually in cloud-provider-azure and it's Nrp related. So I created [a new issue](https://github.com/kubernetes-sigs/cloud-provider-azure/issues/438) in cloud-provider-azure to track it.

**Comment by aramase** [comment]
Thank you for creating the issue. I'll take a look.

**Comment by bridgetkromhout** [comment]
Closing as this issue was moved to cloud-provider-azure; thanks.
